src/routes/write-exercise/WriteExerciseScreen.tsx

```diff
@@ -44,23 +45,31 @@ const WriteExerciseScreen = ({ navigation, route }: WriteExerciseScreenPropsType
   const { trainingSet, trainingSetId, disciplineTitle } = extraParams
   const [isLoading, setIsLoading] = useState(true)
   const [currentDocumentNumber, setCurrentDocumentNumber] = useState(0)
-
-  let documents = useLoadDocuments(trainingSetId).data
+  const [documents, setDocuments] = useState<DocumentType[]>([])
+  const response = useLoadDocuments(trainingSetId)
 
   useEffect(() => {
     AsyncStorage.setSession(route.params).catch(e => console.error(e))
   }, [route.params])
 
-  if (retryData) {
-    documents = retryData.data
-  }
+  useEffect(() => {
+    if (response.data) {
+      setDocuments(response.data)
+    }
+  }, [response.loading])
```

Comment:
the dependencies don't seem to be correct. 
```suggestion
  }, [response])
```